pysph/examples/dam_break_2d.py

```diff
@@ -215,11 +233,30 @@ def create_particles(self):
             nboundary_layers = 4
             nfluid_offset = 1
             wall_hex_pack = False
-        xt, yt = get_2d_tank(dx=self.dx, length=container_width,
-                             height=container_height, base_center=[2, 0],
-                             num_layers=nboundary_layers)
-        xf, yf = get_2d_block(dx=self.dx, length=fluid_column_width,
-                              height=fluid_column_height, center=[0.5, 1])
+
+        if self.compare_data == "buchner":
```

Comment:
Instead of this global hack, which is a vestige of the old example, you can consider a cleaner approach.  Perhaps use a dictionary to declare globally if you want the two setup configurations.  Or you can create a separate case for the Buchner one and leave the 2D dam break to be that of Koshizuka.  So perhaps db_2d_buchner or so?